frmts/gtiff/tif_jxl.c

```diff
@@ -335,7 +335,15 @@ JXLPreDecode(TIFF* tif, uint16_t s)
         format.data_type = jxlDataType;
         format.endianness = JXL_NATIVE_ENDIAN;
         format.align = 0;
-        int bAlphaEmbedded = (info.alpha_bits!=0);
+        // alpha_bits is set even for a gray, gray, Alpha, gray, gray
+        // or for R, G, B, undefined, Alpha
+        // It is only reliable if num_extra_channels == 1
+        // Probably a defect of libjxl
+        // Anyway it is safe to read R,G,B,A,undefined as RGB interleaved
+        // and Alpha and undefined as separate
```

Comment:
Do you have any pointers to this defect ?
From my testing when creating a jxl file like this:
`gdal_translate  -b 1 -b 2 -b 3 -b 4 -b 5 src.tif  -colorinterp red,green,blue,alpha,undefined -co TILED=YES -co COMPRESS=JXL rgbai.tif`
The produced jxl codestream has embedded alpha, but does not pass this added test. Here is the corresponding basic_info:
```bits_per_sample = 16,
num_color_channels = 3,
num_extra_channels = 2, 
alpha_bits = 16,   
alpha_premultiplied = 0
```

Reply:
> Do you have any pointers to this defect ?

I've just reported it as https://github.com/libjxl/libjxl/issues/1773 and added an extra test to detect RGBAxxxx in an explicit way and be able to use the interleaved decoding for the first 4 bands